tests/Api/ValidatorTest.php

```diff
@@ -160,6 +160,33 @@ public function validationProvider()
                 ['foo' => [1, 3]],
                 "Found 1 error while validating the input provided for the Foo operation:\n[foo] must be an associative array. Found array(2)"
             ],
+            // empty array must validate as an assoc array
```

Comment:
Can you add in another test to check for the case of an array with the proper indexed array keys, but out of order? For example:
```
[2 => 'foo', 0 => 'bar', 1 => 'baz']
```